sections/errorhandling/centralizedhandling.md

```diff
@@ -27,24 +27,28 @@ catch (error) {
 
 // Error handling middleware, we delegate the handling to the centralized error handler
 app.use(async (err, req, res, next) => {
-  const isOperationalError = await errorHandler.handleError(err);
-  if (!isOperationalError) {
-    next(err);
-  }
+  await centralizedErrorHandler.handleError(err);
 });
 ```
 
 ### Code example – handling errors within a dedicated object
 
 ```javascript
-module.exports.handler = new errorHandler();
+module.exports.handler = new centralizedErrorHandler();
 
-function errorHandler() {
+function centralizedErrorHandler() {
   this.handleError = async function(err) {
     await logger.logError(err);
     await sendMailToAdminIfCritical;
     await saveInOpsQueueIfCritical;
-    await determineIfOperationalError;
+    const isOperationalError = await determineIfOperationalError(err);
+    if (isOperationalError) {
+      // return an error response to the client
+      return res.status(this.mapErrorToHttp(err));
```

Comment:
I'm not sure this is correct, where is "res" coming from?

Reply:
Now that you mentioned it, I just noticed `res` doesn't come from anywhere…

What would be a good way to solve this? Should it be passed as a parameter to `this.handleError`?

Reply:
I doubt you would want to pass `res` in, as it then ties the function to express. Is the current example not already correct?

```javascript
// Error handling middleware, we delegate the handling to the centralized error handler
app.use(async (err, req, res, next) => {
  const isOperationalError = await errorHandler.handleError(err);
  if (!isOperationalError) {
    next(err);
  }
});
```
The handler performs actions that are common to all errors, then passes off to the caller to handle the specifics e.g. Express would potentially send an appropriate HTTP response, however a non web interface might do something different.